### PR TITLE
Remove static Context from DateUtils, DAOs

### PR DESCRIPTION
**Pull request by frapontillo**
This PR aims at removing static references, especially of `Context`.

There was an annoying reference of static `Context` that was kept to simply retrieve the `TimeZone` from `DateUtils`. The singleton of `DateUtils` has been removed as well, making all methods `static` (it's utility, after all).

### CODE REVIEW

File: app/src/main/java/com/connfa/App.java [hunk 1]
```diff
@@ -19,30 +18,37 @@
...
         DrupalClient client = new DrupalClient(
                 null,
                 Model.instance().createNewQueue(getApplicationContext()),
                 BaseRequest.RequestFormat.JSON,
                 null
         );
+
         DrupalImageView.setupSharedClient(client);
+
+        TwitterAuthConfig authConfig = new TwitterAuthConfig(
```
Comment (rock3r) [inline app/src/main/java/com/connfa/App.java:40]: TODO for next PR, move these with the rest of the API keys :)

File: app/src/main/java/com/connfa/model/AppDatabaseInfo.java [hunk 2]
```diff
@@ -74,14 +74,14 @@ public IMigrationTask getMigrationTask() {
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         if (newVersion > oldVersion) {
-            PreferencesManager.getInstance().saveLastUpdateDate(null);
+            PreferencesManager.create(context).saveLastUpdateDate(null);
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/AppDatabaseInfo.java]: Can we have a `clearLastUpdateTime()` instead?

File: app/src/main/java/com/connfa/model/PreferencesManager.java [hunk 3]
```diff
@@ -11,7 +11,6 @@
 public class PreferencesManager {
 
     private static final String PREF_NAME = "com.ls.drupalconapp.model.MAIN_PREFERENCES";
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/PreferencesManager.java]: We need to take care of these leftovers too (in a next PR)

File: app/src/main/java/com/connfa/model/UpdatesManager.java [hunk 4]
```diff
@@ -35,13 +35,14 @@
     public static final int POIS_REQUEST_ID = 10;
     public static final int INFO_REQUEST_ID = 11;
 
+    private final Context context;
     private DrupalClient mClient;
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/UpdatesManager.java]: Can we make this `final` too?

File: app/src/main/java/com/connfa/model/data/Event.java [hunk 5]
```diff
@@ -25,6 +25,7 @@
     public static final int PROGRAM_CLASS = 1;
     public static final int BOFS_CLASS = 2;
     public static final int SOCIALS_CLASS = 3;
+    private final TimeZone serverTimeZone;
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/data/Event.java:29]: Whiteline before plz

File: app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java [hunk 6]
```diff
@@ -25,31 +32,29 @@
...
      * This method should return list of columns, used to define unique object
      * in "contains" method
-     *
-     * @return
      */
     protected abstract String[] getKeyColumns();
 
-    protected ILAPIDBFacade getFacade() {
-        ILAPIDBFacade dbFacade = LAPIDBRegister.getInstance().lookup(getDatabaseName());
+    protected Context getContext() {
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java]: Make it final plz (this and all the methods that aren't overridden in subclasses)

File: app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java [hunk 7]
```diff
@@ -25,31 +32,29 @@
...
+        return context;
+    }
 
-        return dbFacade;
+    protected ILAPIDBFacade getFacade() {
+        return LAPIDBRegister.getInstance().lookup(getDatabaseName());
     }
 
-    public boolean containsData(ClassToSave theObj) {
+    private boolean containsData(ClassToSave theObj) {
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java]: Can we get out the `the` prefix from params everywhere?

File: app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java [hunk 8]
```diff
@@ -249,20 +250,20 @@ public void saveDataSafe(final ClassToSave theItem) {
         }
     }
 
-    public void saveData(final List<ClassToSave> theList) {
+    public void saveData(List<ClassToSave> theList) {
         for (ClassToSave obj : theList) {
 //            Log.d("saving", obj.toString());
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java]: Remove this plz

File: app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java [hunk 9]
```diff
@@ -1,13 +1,20 @@
 package com.connfa.model.database;
 
 import android.content.ContentValues;
+import android.content.Context;
 import android.database.Cursor;
 
 import java.util.LinkedList;
 import java.util.List;
 
 public abstract class AbstractEntityDAO<ClassToSave extends AbstractEntity<ClassId>, ClassId> {
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/database/AbstractEntityDAO.java]: Can we have a nicer type param name? Such as `E`

File: app/src/main/java/com/connfa/model/database/DatabaseFacade.java [hunk 10]
```diff
@@ -6,8 +6,7 @@
 import android.database.SQLException;
 import android.database.sqlite.SQLiteDatabase;
 
-public class DatabaseFacade
-        implements ILAPIDBFacade {
+class DatabaseFacade implements ILAPIDBFacade {
 //
 //	private static DatabaseFacade instance;
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/database/DatabaseFacade.java]: image

File: app/src/main/java/com/connfa/model/managers/SocialManager.java [hunk 11]
```diff
@@ -72,8 +75,8 @@ protected boolean storeResponse(Event.Holder requestResponse, String tag) {
 //    }
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/managers/SocialManager.java]: image

File: app/src/main/java/com/connfa/ui/fragment/EventFragment.java [hunk 12]
```diff
@@ -188,8 +189,8 @@ private void onItemClick(int position) {
     }
 
 //    private int getCurrentTimePosition(List<EventListItem> eventListItems) {
-//        Calendar calendar = Calendar.getInstance();
-//        calendar.setTimeZone(DateUtils.getInstance().getTimeZone());
+//        Calendar calendar = Calendar.create();
```
Comment (rock3r) [inline app/src/main/java/com/connfa/ui/fragment/EventFragment.java]: image

**Review by rock3r — CHANGES_REQUESTED** [review]
Some changes to have (boyscout!) before shipping but LGTM 👍

File: app/src/main/java/com/connfa/App.java (hunk 1 above)
Comment (frapontillo) [reply app/src/main/java/com/connfa/App.java:40]: It is already a `resValueString`.

**Review by rock3r — APPROVED** [review]

**Pull request merged**